pkg/lib/machines.js

```diff
@@ -26,6 +26,29 @@ function generate_session_key(host) {
     return session_prefix + "/" + host;
 }
 
+export function host_superuser_storage_key(host) {
+    if (!host)
+        host = cockpit.transport.host;
+
+    const local_key = window.localStorage.getItem("superuser-key");
+    if (host == "localhost")
```

Comment:
This is just moved code, not new, but while we are at it: Can it happen that this is "127.0.0.x" or "::1"?

Reply:
No, this is supposed to only catch the primary session, which is always the one with exactly "localhost" as its host value.